packages/react-dom-bindings/src/server/ReactDOMServerExternalRuntime.js

```diff
@@ -13,13 +13,65 @@ import {
   completeSegment,
 } from './fizz-instruction-set/ReactDOMFizzInstructionSet';
 
-// Intentionally does nothing. Implementation will be added in future PR.
-// eslint-disable-next-line no-unused-vars
-const observer = new MutationObserver(mutations => {
-  // These are only called so I can check what the module output looks like. The
-  // code is unreachable.
-  clientRenderBoundary();
-  completeBoundaryWithStyles();
-  completeBoundary();
-  completeSegment();
-});
+// This runtime may be sent to the client multiple times (if FizzServer.render
+//  is called more than once). Here, we check whether the mutation observer
+//  was already created / installed
+if (!window.$REACT_FIZZ_OBSERVER) {
+  // TODO: Eventually remove, we currently need to set these globals for
+  // compatibility with ReactDOMFizzInstructionSet
+  window.$RC = completeBoundary;
```

Comment:
I think we currently access this variable through `window` [here](https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/server/fizz-instruction-set/ReactDOMFizzInstructionSet.js#L52-L53). 

We can add a compile-time variable (similar to `__DEV__`) to compile this access away when bundling the external runtime. There probably is a better solution I'm missing though

Reply:
Yeah I left a TODO for that when I set up the Closure compilation step. I tried setting it up so that the arguments get passed into the function ("dependency injection"), which worked, but Closure wasn't inlining it as well as it should in the inline runtime. I think it could be solved by running Rollup before running Closure, like we do for our main bundles.

I can look into improving that if you like @mofeiZ